### PR TITLE
Added a new line linter

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Added a new-line linter to make sure all files end with a single newline. This is a POSIX standard, read more https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline. Thanks @peternied for introducing it.
 
### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/2857
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/new-line.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Code Hygiene
```
Comment (peternied) [inline .github/workflows/new-line.yml:1]: I'd recommend renaming the file name to match the workflow name, `new-line` -> `code-hygiene`.  In the future you could add addition jobs that scanned for other hygiene related issues.

**Review by peternied — COMMENTED** [review]
Nice!

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@owaiskazi19 I'm not going to backport this into 2.0, I think we can let that one be

**Comment by owaiskazi19** [comment]
> @owaiskazi19 I'm not going to backport this into 2.0, I think we can let that one be

Sounds good. Thanks for the review and taking care of the 2.x backport.

**Pull request merged**